odoo/addons/base/tests/test_expression.py

```diff
@@ -1591,21 +1591,19 @@ def test_empty(self):
         with self.assertQueries(['''
             SELECT "res_partner".id
             FROM "res_partner"
-            WHERE EXISTS (
-                SELECT 1 FROM "res_partner_bank" AS "res_partner__bank_ids"
-                WHERE "res_partner__bank_ids"."partner_id" = "res_partner".id
-            )
+            WHERE ("res_partner"."id" IN (
+                SELECT "partner_id" FROM "res_partner_bank" WHERE "partner_id" IS NOT NULL
+            ))
             ORDER BY "res_partner"."id"
         ''']):
             self.Partner.search([('bank_ids', '!=', False)], order='id')
 
         with self.assertQueries(['''
             SELECT "res_partner".id
             FROM "res_partner"
-            WHERE NOT EXISTS (
-                SELECT 1 FROM "res_partner_bank" AS "res_partner__bank_ids"
-                WHERE "res_partner__bank_ids"."partner_id" = "res_partner".id
-            )
+            WHERE ("res_partner"."id" NOT IN (
+                SELECT "partner_id" FROM "res_partner_bank" WHERE "partner_id" IS NOT NULL
+            ))
```

Comment:
#### Just dumping that here, didn't try the query plan, and didn't try to implement it in `expression.py`


Wouldn't this be more efficient?
```sql
SELECT DISTINCT ON ("res_partner"."id")
       "res_partner"."id"
  FROM "res_partner"
  JOIN "res_partner_bank" ON "res_partner_bank"."partner_id" = "res_partner"."id"
```

As I understand it, current implementation is ~`O(m*n)`, where what I'm proposing is ~`O(m*log(n))`, with `m` the size of `res_partner` and `n` the size of `res_partner_bank`, the difference can be done thanks to indexes.